.github/workflows/java-ci.yml

```diff
@@ -17,10 +17,11 @@ jobs:
         java: [8, 11, 17]
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```

Comment:
Looks like they're up to 4 actually https://github.com/marketplace/actions/checkout should we just go ahead and use that?

Reply:
```suggestion
      - uses: actions/checkout@v4
```